Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -0,0 +1,152 @@
+
+        case .float:
+            return "[\(dictionary) setObject: [NSNumber numberWithDouble:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .integer:
+            return "[\(dictionary) setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .object:
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: [\(propIVarName) dictionaryRepresentation]  forKey: @\"\(param)\" ];"
+                ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                ]
+                })
+                ]}
+        case .string(format: .none),
+             .string(format: .some(.email)),
+             .string(format: .some(.hostname)),
+             .string(format: .some(.ipv4)),
+             .string(format: .some(.ipv6)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.uri)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: ["+propIVarName + " absoluteString] forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.dateTime)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: [[NSValueTransformer valueTransformerForName:\(dateValueTransformerKey)] transformedValue:\(propIVarName)]  forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .enumT(.integer):
+            return "[\(dictionary) setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .enumT(.string):
+            return "[\(dictionary) setObject: "+enumToStringMethodName(propertyName: param, className: self.className) + "(\(propIVarName))" + " forKey: @\"" + param + "\" ];"
+        case .array(itemType: let itemType?):
+            let currentResult = "result\(counter)"
+            let currentObj = "obj\(counter)"
+            switch itemType {
+            case .reference,
```

Comment:
You may not even need this switch statement if your implementation of the default case is recursive since states like object and reference will be handled.